### PR TITLE
docs(components): use `""` replace null for no icon setting

### PR DESCRIPTION
**Pull request by thinkasany**
修复icon的type

image

image

### CODE REVIEW

File: packages/components/page-header/src/page-header.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+import type { Component, ExtractPropTypes } from 'vue'
 import type PageHeader from './page-header.vue'
 
 export const pageHeaderProps = buildProps({
   /**
    * @description icon component of page header
    */
   icon: {
-    type: iconPropType,
+    type: definePropType<string | Component | null>([String, Object, Function]),
```
Comment (btea) [inline packages/components/page-header/src/page-header.ts:11]: Maybe you should add a `withNullIconPropType` in the same file as the `iconPropType` type.

**Comment by btea** [comment]
This is a legacy issue. Setting the icon to an empty string should also achieve the desired effect. Perhaps we should change the documentation.

File: packages/components/page-header/src/page-header.ts (hunk 1 above)
Comment (thinkasany) [reply packages/components/page-header/src/page-header.ts]: done.

**Comment by thinkasany** [comment]
> This is a legacy issue. Setting the icon to an empty string should also achieve the desired effect. Perhaps we should change the documentation.

what I need to do, could you please elaborate a little bit?

**Review by warmthsea — COMMENTED** [review]
Here we also need.

https://github.com/element-plus/element-plus/blob/ead10025339937e584f1814d19418332b48d897a/docs/examples/page-header/additional-sections.vue#L2

**Comment by thinkasany** [comment]
> Here we also need.
> 
> https://github.com/element-plus/element-plus/blob/ead10025339937e584f1814d19418332b48d897a/docs/examples/page-header/additional-sections.vue#L2

done.

**Review by warmthsea — APPROVED** [review]

**Review by btea — APPROVED** [review]
Thanks!

**Pull request merged**